### PR TITLE
Make InternalModel and Resnet work with rexportable flow

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Summary: Internal model and Resnet uses "re-export" flow now. Also did some refactoring to make the code little cleaner

Some changes for OSS:
1. Correctly use the "cached" fake tensors so that static symbols are still resolved to static 
2. Change logic in PassBase to allocate static shapes for parameters 
3. Add "is_torch_exported" tag to every node to make it survive during various graph transformations.
4. Added experimental wrapper API for quantization team to get pre_dispatch=True graph. Note that it doesn't actually do that right now. But we plan to switch soon. 

Test Plan: CI

Differential Revision: D47890878

### CODE REVIEW

File: torch/_export/__init__.py
Comment (jerryzh168) [inline torch/_export/__init__.py:130]: can you add some docs about plans for this API

Comment (tugsbayasgalan) [reply torch/_export/__init__.py:133]: just added

**Review by jerryzh168 — APPROVED** [review]
quant changes looks good to me, do you need review from some compiler people as well?

**Comment by tugsbayasgalan** [comment]
> quant changes looks good to me, do you need review from some compiler people as well?

Sounds good! I will merge after I get review from Zhengxu.

**Pull request closed without merging**
